apps/hermes/client/js/src/HermesConnection.ts

```diff
@@ -0,0 +1,198 @@
+import { EncodingType, PriceFeedMetadata, PriceUpdate } from "./types";
+import EventSource from "eventsource";
+
+const DEFAULT_TIMEOUT: DurationInMs = 5000;
+const DEFAULT_HTTP_RETRIES = 3;
+
+export type UnixTimestamp = number;
+export type DurationInSeconds = number;
+export type HexString = string;
+export type DurationInMs = number;
+
+export type HermesConnectionConfig = {
+  /* Timeout of each request (for all of retries). Default: 5000ms */
+  timeout?: DurationInMs;
+  /**
+   * Number of times a HTTP request will be retried before the API returns a failure. Default: 3.
+   *
+   * The connection uses exponential back-off for the delay between retries. However,
+   * it will timeout regardless of the retries at the configured `timeout` time.
+   */
+  httpRetries?: number;
+};
+
+export class HermesConnection {
+  private baseURL: string;
+  private timeout: DurationInMs;
+  private httpRetries: number;
+
+  /**
+   * Constructs a new Connection.
+   *
+   * @param endpoint endpoint URL to the price service. Example: https://website/example/
+   * @param config Optional HermesConnectionConfig for custom configurations.
+   */
+  constructor(endpoint: string, config?: HermesConnectionConfig) {
+    this.baseURL = endpoint;
+    this.timeout = config?.timeout || DEFAULT_TIMEOUT;
+    this.httpRetries = config?.httpRetries || DEFAULT_HTTP_RETRIES;
+  }
+
+  private async httpRequest<ResponseData>(
+    url: string,
+    options?: RequestInit,
+    retries = this.httpRetries,
+    backoff = 300
+  ): Promise<ResponseData> {
```

Comment:
If you follow my advice above about using a schema validator, then rather than casting, here you would pass a schema and add a schema parse before returning the result.  Then you don't have to typecast unsafely.